doc/pmon/pmon-sensormon.md

```diff
@@ -0,0 +1,181 @@
+# SONiC PMON Sensor Monitoring Enhancement #
+
+
+### Revision 1.0 
+
+## Table of Content 
+1. [Scope](#Scope)
+2. [Definitions](#Definitions/Abbreviations)
+3. [Overview](#Overview)
+3. [Requirements](#Requirements)
+4. [High Level Design](#High-Level-Design)
+5. [CLI](#CLI-Enhancements)
+7. [Test](#Testing-Considerations)
+
+
+
+### Scope  
+
+This document covers the support for monitoring voltage and current sensor devices in SONiC. 
+
+### Definitions/Abbreviations 
+
+PMON - Platform Monitor container in SONiC.
+
+PSU - Power Supply Unit
+
+Voltage Sensor - Sensor device which can report a voltage measurement in the system.
+
+Current Sensor - Sensor device which can report current measurement in the system.
+
+Altitude Sensor - Sensor device which can report the altitude of the system.
+
+
+### Overview 
+
+Modern hardware systems have many different types of sensors and control devices. Voltage sensor devices can measure and in some cases control the voltages on the boards. Current sensor devices can measure current. It is also possible to have other types of sensors such as Altitude Sensors etc. These devices can report measurements from different parts of the system which are useful for monitoring system health. For example, voltage controller devices distribute power across different parts of the system such as  motherboard, daughterboards etc. and can report voltage measurements from there. Often these devices can report under-voltage/over-voltage faults which should be monitored to alert the operator about any power related failures in the system. This document provides an overview for monitoring voltage and current sensors in SONiC. The solution proposed in this document can be enhanced for other types of sensors as well.
+
+Note that temperature sensor devices are managed via SONiC ThermalCtlD daemon today. At this point there is no change proposed for ThermalCtlD. This proposed design can be used for voltage, current and other types of sensors.
+
+Linux does provide some support of voltage and current sensor monitoring using lmsensors/hwmon infrastructure. However there are a few limitations with that
+
+- Devices not supported with Hwmon are not covered
```

Comment:
@bmridul could you list few such devices not supported by hwmon? Why not we add support for them in hwmon?

Reply:
I donot have a comprehensive list of devices. I had looked at the hardware designs in our company and found some devices for which the standard drivers did not include hwmon support. This is what I replied in the [comment](https://github.com/sonic-net/SONiC/pull/1394#issuecomment-1638528187) earlier.  A bunch of devices seem to support Linux IIO infra and donot include the hwmon/lmsensors support.

https://github.com/torvalds/linux/tree/master/drivers/iio/adc
https://github.com/torvalds/linux/tree/master/drivers/iio/dac

These drivers are contributed from the device vendors and adding hwmon support will need us to co-ordinate with different vendors and would take a long time. 
.